src/training/graph_group_sync.h

```diff
@@ -23,6 +24,9 @@ class SyncGraphGroup : public GraphGroup, public ExponentialSmoothing {
   std::vector<Ptr<TensorAllocator>> paramsAllocs_; // [deviceIndex] we must hold a reference to the memory until this class dies
   // @TODO: move this nto ExponentialSmoothing, together with paramsAvg_?
 
+  // model compressor
+  std::vector<Ptr<Compresser>> compressers_;
```

Comment:
If quantization currently works only for `SyncGraphGroup`, there should be a warning/abort if somebody tries to use quantization options without `--sync-sgd`.

Reply:
We should consider making `--sync-sgd` the default actually. We are currently carrying to many old defaults around that we adopted from 2016's Nematus. Time to clean that up a bit.

Reply:
But notwithstanding that, you are right about the abort for the current code.

Reply:
abort is added